### DISCUSSION TITLE
Data Validation Engine - add support for custom validators via GIT Repositories Integration

### DISCUSSION THREAD

**Discussion by TheBirdsNest** [nautobot/nautobot#6243] [Plugin (App) Ideas & Development]
As a Network Engineer, I'd like to be able to apply policies/restrictions to configuration in Nautobot easily with the use of Git. 

Many validation steps we want to take, are not supported with the current out-of-the-box plugin. 
For example, 

I want to restrict the VLAN ID someone can select, based on the selected role (if role = 'Users', VID must be between >=10, <=20).
For this a custom validator can be written, but I want to be able to create/update and manage these through the Nautobot Git Repository sync feature and not by manually changing files (or in my case rebuilding containers) that also require application restarts.

In Git Repositories, a new "Provides" would exist that includes "Data Validators".

Example Validator that can be managed through Git:

custom_rules/vlans.py:
```
from nautobot.dcim.models import VLAN
from nautobot_data_validation_engine.plugins import BaseValidator, register_validator

class VLANRoleVIDValidator(BaseValidator):
    model = VLAN
    field = "vid"

    def validate(self, instance):
        if instance.role and instance.role.name == "User Access":
            if not (20 >= instance.vid <= 29):
                self.fail(
                    instance,
                    f"VLAN ID {instance.vid} is not within the allowed range 20-29 for role 'User Access'."
                )

register_validator(VLANRoleValidator)
```

**Comment by itdependsnetworks** [comment]
Seems reasonable to me, not sure where this would get prioritized, but in general we use and should be supportive of this pattern. 

I would suspect a PR would be accepted, you could use [this](https://github.com/nautobot/nautobot-app-golden-config/pull/206) and [follow up](https://github.com/nautobot/nautobot-app-golden-config/pull/209) as guides.
